jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -135,6 +135,127 @@ ecma_builtin_object_object_get_prototype_of (ecma_value_t this_arg, /**< 'this'
   return ret_value;
 } /* ecma_builtin_object_object_get_prototype_of */
 
+#ifndef CONFIG_DISABLE_ES2015_BUILTIN
```

Comment:
Motivation for this name came from `CONFIG_DISABLE_ANNEXB_BUILTIN`, which is a catch-all guard for miscellaneous extensions to the ES5.1 standard. They are not specific to a single object but add things "here-and-there". This guard `CONFIG_DISABLE_ES2015_BUILTIN` could do the same.

In some discussions, there was an idea that we could keep this guard as is and
1. change `CONFIG_DISABLE_ARRAYBUFFER_BUILTIN` to `CONFIG_DISABLE_ES2015_ARRAYBUFFER_BUILTIN` and `CONFIG_DISABLE_TYPEDARRAY_BUILTIN` to `CONFIG_DISABLE_ES2015_TYPEDARRAY_BUILTIN`, and
2. make all `CONFIG_DISABLE_ES2015_xxx_BUILTIN` guards dependent on this `CONFIG_DISABLE_ES2015_BUILTIN` guard (so, if ES2015 is disabled in general then there will be no need for disabling all ES2015-specific builtins one by one).

(But this would happen in a follow-up commit.)
Would that work?